archunit/src/test/java/com/tngtech/archunit/core/domain/JavaPackageTest.java

```diff
@@ -241,77 +241,151 @@ public void visits_packages() {
     }
 
     @Test
-    public void has_class_dependencies_to_other_packages() {
+    public void has_class_dependencies_from_this_package() {
         JavaPackage examplePackage = importPackage("packageexamples");
 
-        assertThatDependencies(examplePackage.getPackage("second").getClassDependenciesFromSelf())
+        assertThatDependencies(examplePackage.getPackage("second").getClassDependenciesFromThisPackage())
+                .contain(Second1.class, First2.class)
+                .doesNotContain(ClassDependingOnOtherSecondClass.class, Second1.class)
```

Comment:
I'm not sure I followed this.

Do you mean that if we added `second.Second2` that depends on `second.sub.SecondSub1`, this dependency should be returned from `getClassDependenciesFromThisPackage` but NOT from `getClassDependenciesFromThisPackageAndSubpackages`?
The current code would return it from both these methods.

Reply:
Hmm, I would have thought, that this example should not be contained in the dependencies, because from the view of `thisPackageAndSubpackages` this is a self-dependency, right? :thinking: I.e. why filter out dependencies where origin or target are in the root package, but not for subpackages? :thinking: I think that was an original oversight and should not be that way, I'm gonna fix it...

Reply:
I fixed it and rebased your commit on top. So please check if that makes sense. But I think this was broken before, if I say "I consider all classes in the tree" for dependencies, then I should also treat self-references with respect to that set. Why would `mypkg.A -> mypkg.B` not be contained, but `mypkg.A -> mypkg.sub.B` would be? :man_shrugging: Anyway, let me know what you think about that...

Reply:
Agreed, makes sense.